src/core_plugins/input_control_vis/public/vis_controller.js

```diff
@@ -28,7 +38,31 @@ class VisController {
     unmountComponentAtNode(this.el);
   }
 
+  isHorizontalLayout(panelWidth, panelHeight) {
+    return panelWidth > (1.5 * panelHeight) || panelWidth < 200;
+  }
+
   drawVis() {
+    // unable to use this.el dimensions because this.el.clientWidth does not shrink
+    // once controlWidth has been set to a large number
+    const panelNode = this.getPanelNode();
+    if (!panelNode) {
+      throw new Error('Unable to find panel node in ancestor tree');
+    }
+    const panelWidth = panelNode.clientWidth;
+    const panelHeight = panelNode.clientHeight;
+    // todo - how do we make this less fragile to css changes?
```

Comment:
maybe we need something exposed to the embeddable from dashboard - width/height that is only the inner values, and an onResize function.

Reply:
I filed https://github.com/elastic/kibana/issues/14921 so I can keep track of these kinds of things.  Plus then maybe we can remember to go back and clean this up, since I agree, it's going to be very fragile.  

What about adding a selenium test to try to ensure we don't break it accidentally?  I'm dubious whether the react test would catch any issues.  I've had problems testing dimension values in the jest tests because jsdom returns all 0 for getClientBoundingRect (or whatever that function is called).